src/app/util/client-api.cpp

```diff
@@ -210,7 +210,8 @@ static uint16_t vFillBuffer(uint8_t * buffer, uint16_t bufferLen, uint8_t frameC
         // The destination buffer must be at least as large as the running total
         // plus the length of the integer value (if applicable) plus the length of
         // the data (if applicable).
-        if (bufferLen < bytes + dataLen + valueLen)
+        // Fix #19351 - Cast up to larger type during summation to prevent integer overflow issues.
+        if (bufferLen < (size_t) bytes + (size_t) dataLen + (size_t) valueLen)
```

Comment:
This doesn't look right.  Won't bytes / dataLen / valueLen be promoted to integer width for addition on systems with 32-bit-or-wider integers?  So the breakage is on systems with 16-bit integers.  BUT size_t can also be 16-bit, so the size_t cast may actually not fix this.

Reply:
Alternatively, if I am misunderstanding the promotion rules and this does fix a real problem on 32-bit systems, it still seems like security fixes like this should not be made in such a way that the fix explicitly does not work on smaller systems -> i.e. the size_t cast still seems like a bad fix.

Reply:
I agree with you that the recommendation to use `size_t` isn't a good choice.  How about `uint32_t` instead?

While the recommended code was `size_t`, the text just stated `cast to a larger type`, so `uint32_t` should work for systems large and small:
"We recommend the overflow be checked with help of cast to a larger type. While casting the leftmost summand variable will be enough due to the type promotion rule, it might be less prone to future errors to just cast them all."

Reply:
Yes, I think uint32_t covers all possible cases, including unlikely platforms with 16-bit size_t.